internal/entities/properties/service/service.go

```diff
@@ -331,3 +336,21 @@ func (ps *propertiesService) EntityWithProperties(
 
 	return models.NewEntityWithProperties(ent, props), nil
 }
+
+// EntityWithPropertiesAsProto converts the entity with properties to a protobuf message
+func (_ *propertiesService) EntityWithPropertiesAsProto(
+	ctx context.Context, ewp *models.EntityWithProperties, provMgr manager.ProviderManager,
```

Comment:
We could make the method more generic and instead of requiring EntityWithProperties let he caller just pass the provider ID and entity type